src/Razor/test/Microsoft.AspNetCore.Razor.Test.ComponentShim/Microsoft.AspNetCore.Components/EventCallbackFactory.cs

```diff
@@ -18,7 +18,7 @@ public sealed class EventCallbackFactory
 
         public EventCallback Create(object receiver, Func<object, Task> callback) => default;
 
-        public EventCallback<T> Create<T>(object receiver, string callback) => default;
+        public string Create<T>(object receiver, string callback) => default;
```

Comment:
But why?!!

Reply:
Because eventhandler attributes are not syntactically distinct when they are JS and when they are C#.

```C#
<button onclick="@foo" />
```

Is it JS or C#? Well that depends on the type of `@foo` - which we don't know at codegen-time.